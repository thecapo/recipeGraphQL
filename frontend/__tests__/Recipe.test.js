import RecipeComponent from '../components/Recipe';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeRecipe = {
  title: 'Delicious Food',
  ingredients: '3tsbp chocolate, 2/4cup water',
  instructions: 'Bacon is the water of all meat',
  image: 'food.jpg',
  largeImage: 'food-large.jpg',
};

describe('<Recipe/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<RecipeComponent recipe={fakeRecipe} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    const wrapper = shallow(<RecipeComponent recipe={fakeRecipe} />);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeRecipe.image);
    expect(img.props().alt).toBe(fakeRecipe.title);
  });

  it('renders the title and number of ingredients', () => {
    const wrapper = shallow(<RecipeComponent recipe={fakeRecipe} />);
    const list = wrapper.find('ul')
    expect(wrapper.find('Title a').text()).toBe(fakeRecipe.title);
    expect(list.children().length).toBe(2);
  });

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<RecipeComponent recipe={fakeRecipe} />);
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link')).toHaveLength(2);
    expect(buttonList.find('DeleteRecipe').exists()).toBe(true);
  })
})

