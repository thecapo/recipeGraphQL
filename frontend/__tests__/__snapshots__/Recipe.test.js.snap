// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipe/> renders and matches the snapshot 1`] = `
<RecipeStyles>
  <img
    alt="Delicious Food"
    src="food.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/recipe",
          "query": Object {
            "id": undefined,
          },
        }
      }
    >
      <a>
        Delicious Food
      </a>
    </Link>
  </Title>
  <ul>
    <li
      key="0"
    >
      3tsbp chocolate
    </li>
    <li
      key="1"
    >
      2/4cup water
    </li>
  </ul>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/recipe",
          "query": Object {
            "id": undefined,
          },
        }
      }
    >
      <a>
        Read Recipe
      </a>
    </Link>
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": undefined,
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <DeleteRecipe>
      Delete Recipe
    </DeleteRecipe>
  </div>
</RecipeStyles>
`;
